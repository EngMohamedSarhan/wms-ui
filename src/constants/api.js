export const BASE_URL = `${process.env.REACT_APP_BASE_URL}/api/v1`;

export const WAREHOUSE_LOCATIONS_API = "/lookup/warehouse-locations";

export const LPN_DATE_CHECK_API = "/lpndatecheck";

export const ADJUST_ONE_API = "/lpndatecheck/save-single-item";

export const ADJUST_ALL_API = "/lpndatecheck/save-all";

export const APPLICATIONS_API = "/lookup/applications";
