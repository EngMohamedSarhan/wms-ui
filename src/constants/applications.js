import { LPN_DATE_CHECK_ROUTE, TASK_GROUP_ROUTE } from "./routes";

const applications = [
  {
    applicationId: 1,
    applicationName: "iLPN Date Check",
    descriptionTxt: "Random description",
    applicationRoute: LPN_DATE_CHECK_ROUTE,
  },
  {
    applicationId: 2,
    applicationName: "Task Group Eligibility",
    descriptionTxt: "Random description",
    applicationRoute: TASK_GROUP_ROUTE,
  },
];

export default applications;
