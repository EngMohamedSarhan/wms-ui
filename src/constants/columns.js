const columns = [
  { id: "item", label: "Item", minWidth: 70 },
  { id: "itemDescription", label: "Item Description", minWidth: 150 },
  { id: "lpnId", label: "LPN ID", minWidth: 120 },
  {
    id: "lpnStatusDescription",
    label: "LPN Status Description",
    minWidth: 120,
  },
  { id: "displayLocation", label: "Display Location", minWidth: 70 },
  {
    id: "lpnQuantity",
    label: "LPN Quantity In Cases",
    minWidth: 120,
    format: (value) => value.toFixed(2),
  },
  { id: "reason", label: "Reason", minWidth: 120 },
  { id: "dateCode", label: "Date Code", minWidth: 70 },
  {
    id: "manufacturingDate",
    label: "Item Track Manufacturing Date",
    minWidth: 100,
  },
  { id: "expiryDate", label: "Item Track Expiry Date", minWidth: 100 },
  { id: "manufactureDateDiff", label: "Manufacture Date Diff", minWidth: 100 },
  { id: "expirationDateDiff", label: "Expiration Date Diff", minWidth: 100 },
  {
    id: "manufacturedDate",
    label: "Manufactured Date",
    minWidth: 200,
    isDate: true,
  },
  {
    id: "expirationDate",
    label: "Expiration Date",
    minWidth: 200,
    isDate: true,
  },
  {
    id: "consumptionPriorityDate",
    label: "Consumption Priority Date",
    minWidth: 200,
    isDate: true,
  },
  { id: "save", isSave: true, minWidth: 40 },
];

export default columns;
