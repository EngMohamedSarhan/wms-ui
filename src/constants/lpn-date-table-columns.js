const columns = [
  { id: "item", label: "Item", minWidth: 100 },
  { id: "itemDescription", label: "Item Description", minWidth: 150 },
  { id: "lpnId", label: "LPN ID", minWidth: 120 },
  {
    id: "lpnStatusDescription",
    label: "LPN Status Description",
    minWidth: 150,
  },
  { id: "displayLocation", label: "Display Location", minWidth: 100 },
  {
    id: "lpnQuantity",
    label: "LPN Quantity In Cases",
    minWidth: 180,
    format: (value) => value.toFixed(2),
  },
  { id: "reason", label: "Reason", minWidth: 160 },
  { id: "dateCode", label: "Date Code", minWidth: 100 },
  {
    id: "manufacturingDate",
    label: "Item Track Manufacturing Date",
    minWidth: 180,
  },
  { id: "expiryDate", label: "Item Track Expiry Date", minWidth: 140 },
  { id: "manufactureDateDiff", label: "Manufacture Date Diff", minWidth: 160 },
  { id: "expirationDateDiff", label: "Expiration Date Diff", minWidth: 130 },
  {
    id: "manufacturedDate",
    label: "Manufactured Date",
    minWidth: 200,
    isManufactureDate: true,
  },
  {
    id: "expirationDate",
    label: "Expiration Date",
    minWidth: 200,
    isExpirationDate: true,
  },
  {
    id: "consumptionPriorityDate",
    label: "Consumption Priority Date",
    minWidth: 230,
    isPriorityDate: true,
  },
  { id: "save", isSave: true, minWidth: 40 },
];

export default columns;
