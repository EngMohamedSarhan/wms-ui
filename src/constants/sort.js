export const SortOrders = {
  desc: "desc",
  asc: "asc",
};
