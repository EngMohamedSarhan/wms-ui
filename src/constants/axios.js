import axiosInstance from "axios";
import Cookies from "js-cookie";

import { DEVELOPMENT_API } from "./api";
import { TOKEN_COOKIE } from "./cookies";

const axios = axiosInstance.create({
  baseURL: DEVELOPMENT_API,
  timeout: 5000,
});

(async () => {
  axios.defaults.baseURL = DEVELOPMENT_API;
  const token = Cookies.get(TOKEN_COOKIE);

  if (token) {
    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
  }
})();

export default axios;
