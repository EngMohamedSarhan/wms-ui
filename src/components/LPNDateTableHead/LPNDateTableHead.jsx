import {
  Grid,
  IconButton,
  TableCell,
  TableHead,
  TableRow,
  TextField,
} from "@mui/material";
import PropTypes from "prop-types";
import React from "react";

import RotatableArrowUpward from "../../components-styled/RotatableArrowUpward/RotatableArrowUpward";
import columns from "../../utilities/constants/lpn-date-table-columns";
import { SortOrders } from "../../utilities/constants/sort";

const LPNDateTableHead = ({
  sort,
  unsavedRowsRef,
  setSort,
  sortOrder,
  setSortOrder,
  filters,
  setFilters,
  onFilter,
  onChangesModalOpen,
}) => {
  const handleResetSort = () => {
    setSort(null);
    setSortOrder(SortOrders.asc);
  };

  const handleSort = (id) => {
    if (Object.keys(unsavedRowsRef.current).length !== 0) {
      onChangesModalOpen();
      return;
    }

    if (sort !== id) {
      setSort(id);
      setSortOrder(SortOrders.asc);
    } else if (sortOrder === SortOrders.asc) {
      setSortOrder(SortOrders.desc);
    } else {
      handleResetSort();
    }
  };

  const handleFilterChange = (e, id) => {
    if (Object.keys(unsavedRowsRef.current).length !== 0) {
      onChangesModalOpen();
      return;
    }

    if (!e.target.value) {
      delete filters[id];
      setFilters({ ...filters });
    } else {
      filters = { ...filters, [id]: e.target.value };
      setFilters(filters);
    }
    onFilter(filters);
  };

  return (
    <TableHead>
      <TableRow>
        {columns.map((column) => (
          <TableCell
            key={column.id}
            align={column.align}
            sx={{
              minWidth: column.minWidth,
              fontSize: "16px",
              color: "primary.main",
              pt: 0.5,
              p: 1,
            }}
          >
            {!column.isSave ? (
              <Grid container flexDirection="column">
                <Grid container>
                  <Grid item xs={8} sx={{ height: 75 }}>
                    {column.label}
                  </Grid>
                  <Grid item xs={4}>
                    <IconButton
                      color={sort === column.id ? "primary" : undefined}
                      size="small"
                      sx={{ marginLeft: "2px", width: 24, height: 24 }}
                      onClick={() => handleSort(column.id)}
                    >
                      <RotatableArrowUpward
                        fontSize="small"
                        open={sort === column.id && sortOrder === SortOrders.desc}
                      />
                    </IconButton>
                  </Grid>
                </Grid>
                <TextField
                  size="small"
                  inputProps={{ sx: { p: 0.2, pl: 0.5, pr: 0.5 } }}
                  value={filters[column.id] || ""}
                  onChange={(e) => handleFilterChange(e, column.id)}
                />
              </Grid>
            ) : null}
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

LPNDateTableHead.propTypes = {
  sort: PropTypes.string,
  unsavedRowsRef: PropTypes.object.isRequired,
  setSort: PropTypes.func.isRequired,
  sortOrder: PropTypes.string.isRequired,
  setSortOrder: PropTypes.func.isRequired,
  filters: PropTypes.object.isRequired,
  setFilters: PropTypes.func.isRequired,
  onFilter: PropTypes.func.isRequired,
  onChangesModalOpen: PropTypes.func.isRequired,
};

export default LPNDateTableHead;
