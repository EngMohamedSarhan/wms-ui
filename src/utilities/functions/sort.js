import { SortOrders } from "../constants/sort";

export const descendingComparator = (a, b, orderBy) => {
  if (b[orderBy] < a[orderBy] || (!b[orderBy] && a[orderBy])) {
    return -1;
  }
  if (b[orderBy] > a[orderBy] || (!a[orderBy] && b[orderBy])) {
    return 1;
  }
  return 0;
};

export const getComparator = (order, orderBy) =>
  order === SortOrders.desc
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
