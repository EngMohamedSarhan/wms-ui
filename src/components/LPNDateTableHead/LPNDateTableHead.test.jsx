/* eslint-disable no-undef */
import { fireEvent, render } from "@testing-library/react";
import React from "react";

import { SortOrders } from "../../utilities/constants/sort";
import LPNDateTableHead from "./LPNDateTableHead";

describe("LPNDateTableHead Component", () =>
  test("LPNDateTableHead action", () => {
    const setSort = jest.fn();
    const setSortOrder = jest.fn();
    const setFilters = jest.fn();
    const onFilter = jest.fn();
    const { container } = render(
      <table>
        <LPNDateTableHead
          setSort={setSort}
          sortOrder={SortOrders.asc}
          setSortOrder={setSortOrder}
          filters={{}}
          setFilters={setFilters}
          onFilter={onFilter}
        />
      </table>
    );
    const sortButton = container.getElementsByTagName("button")[0];
    fireEvent.click(sortButton);
    expect(setSort).toBeCalled();
    expect(setSortOrder).toBeCalled();
    const filterInput = container.getElementsByTagName("input")[0];
    fireEvent.change(filterInput, { target: { value: "a" } });
    expect(setFilters).toBeCalled();
    expect(onFilter).toBeCalled();
  }));
