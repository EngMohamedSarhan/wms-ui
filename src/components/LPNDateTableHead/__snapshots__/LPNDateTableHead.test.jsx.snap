// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LPNDateTableHead Component LPNDateTableHead snapshot 1`] = `
<ForwardRef(TableHead)>
  <ForwardRef(TableRow)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 90,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Item
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 220,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Item Description
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 110,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            LPN ID
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 145,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            LPN Status Description
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 95,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Display Location
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 160,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            LPN Quantity In Cases
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 180,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Manufactured Date
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 180,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Expiration Date
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 210,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Consumption Priority Date
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 180,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Suggested CPD
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 110,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Product Life In Days
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 150,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Reason
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 90,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Date Code
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 175,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Item Track Manufacturing Date
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 140,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Item Track Expiry Date
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 160,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Manufacture Date Diff
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 130,
          "p": 1,
          "pt": 0.5,
        }
      }
    >
      <ForwardRef(Grid)
        container={true}
        flexDirection="column"
      >
        <ForwardRef(Grid)
          container={true}
        >
          <ForwardRef(Grid)
            item={true}
            sx={
              Object {
                "height": 75,
              }
            }
            xs={8}
          >
            Expiration Date Diff
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            <ForwardRef(IconButton)
              onClick={[Function]}
              size="small"
              sx={
                Object {
                  "height": 24,
                  "marginLeft": "2px",
                  "width": 24,
                }
              }
            >
              <Styled(memo)
                fontSize="small"
                open={false}
              />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(TextField)
          inputProps={
            Object {
              "sx": Object {
                "p": 0.2,
                "pl": 0.5,
                "pr": 0.5,
              },
            }
          }
          onChange={[Function]}
          size="small"
          value=""
        />
      </ForwardRef(Grid)>
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "primary.main",
          "fontSize": "16px",
          "minWidth": 40,
          "p": 1,
          "pt": 0.5,
        }
      }
    />
  </ForwardRef(TableRow)>
</ForwardRef(TableHead)>
`;
